### PR TITLE
fix: open api generated java rest api code

### PR DESCRIPTION
**Pull request by charan678**
# Fixes #

1. Code compilation of autogenerated java api code
2. Wrote basic test using mockito ( This lib will be replacement of jmockito)
3. Updated prism.sh script to run both golang and java unit test in parallel
4. Wrote Inflector to handle plural to singular conversion ( removed open source library inflector)
5. Some test cases are ignore, which will be handled during upcoming stories
6. exclude some test cases in java.Dockerfile because these test cases are using rest api reference. Will create different story to handle these test cases
7. exclude examples folder in java.Dockerfile and will be handled in different story.
8. IncomePhoneNumber model is skipped as nested properties is not handled in the current functionality.
9. Existing Issues: package name is not in proper format, Domains.java is in wrong folder, handle Enum nested types

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 1]
```diff
@@ -237,6 +250,9 @@ private void flattenStringMap(final Map<String, Object> resource, final String k
     }
 
     private String getResourceName(final String path) {
+        if (inflector.isAbbrevation(PathUtils.getLastPathPart(PathUtils.cleanPath(path)))) {
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:253]: nit: should save off the result to `cleanPath` and `getLastPathPart` so they don't need to be called multiple times.

Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:253]: nit: should save off the result to `cleanPath` and `getLastPathPart` so they don't need to be called multiple times.

**Review by childish-sambino — APPROVED** [review]

File: src/main/java/com/twilio/oai/Inflector.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.regex.Pattern;
+
+
+public class Inflector {
+
+    ExpressionRule ruleBook = new ExpressionRule();
+
+    public  String singularize(String word) {
+        if (word == null) return null;
+        String wordStr = word.toString().trim();
```
[the remark this answers is not in the document]
Comment (charan678) [reply src/main/java/com/twilio/oai/Inflector.java]: resolved

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 1 above)
Comment (charan678) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java]: resolved

**Pull request merged**